Source/WebCore/accessibility/AccessibilityNodeObject.cpp

```diff
@@ -1825,11 +1825,46 @@ void AccessibilityNodeObject::alternativeText(Vector<AccessibilityText>& textOrd
             textOrder.append(AccessibilityText(accessibleNameForNode(*object->node()), AccessibilityTextSource::Alternative));
     }
 
-    // The figure element derives its alternative text from the first associated figcaption element if one is available.
-    if (isFigureElement()) {
-        AccessibilityObject* captionForFigure = this->captionForFigure();
-        if (captionForFigure && !captionForFigure->isHidden())
-            textOrder.append(AccessibilityText(accessibleNameForNode(*captionForFigure->node()), AccessibilityTextSource::Alternative));
+    if (auto* image = dynamicDowncast<HTMLImageElement>(*node); image && objectCache) {
```

Comment:
I would remove this null check of `objectCache` since the check is so far removed from where we actually dereference it. It would be better to null check it closer to the first dereference.